src/stdlib_linalg_eigenvalues.fypp

```diff
@@ -0,0 +1,579 @@
+         k   = min(m,n)
+
+         !> Allocate return storage
+         allocate(lambda(k))
+
+         !> Compute eigenvalues only
+         call stdlib_linalg_eig_${ri}$(amat,lambda,overwrite_a=.false.,err=err)
+
+     end function stdlib_linalg_eigvals_${ri}$
+
+     module function stdlib_linalg_eigvals_noerr_${ri}$(a) result(lambda)
+     !! Return an array of eigenvalues of matrix A.
+         !> Input matrix A[m,n]
+         ${rt}$, intent(in), target :: a(:,:)
+         !> Array of eigenvalues
+         complex(${rk}$), allocatable :: lambda(:)
+
+         !> Create
+         ${rt}$, pointer :: amat(:,:)
+         integer(ilp) :: m,n,k
+
+         !> Create an internal pointer so the intent of A won't affect the next call
+         amat => a
+
+         m   = size(a,1,kind=ilp)
+         n   = size(a,2,kind=ilp)
+         k   = min(m,n)
+
+         !> Allocate return storage
+         allocate(lambda(k))
+
+         !> Compute eigenvalues only
+         call stdlib_linalg_eig_${ri}$(amat,lambda,overwrite_a=.false.)
+
+     end function stdlib_linalg_eigvals_noerr_${ri}$
+
+     module subroutine stdlib_linalg_eig_${ri}$(a,lambda,right,left,overwrite_a,err)
+     !! Eigendecomposition of matrix A returning an array `lambda` of eigenvalues, 
+     !! and optionally right or left eigenvectors.        
+         !> Input matrix A[m,n]
+         ${rt}$, intent(inout), target :: a(:,:)
+         !> Array of eigenvalues
+         complex(${rk}$), intent(out) :: lambda(:)
+         !> [optional] RIGHT eigenvectors of A (as columns)
+         complex(${rk}$), optional, intent(out), target :: right(:,:)
+         !> [optional] LEFT eigenvectors of A (as columns)
+         complex(${rk}$), optional, intent(out), target :: left(:,:)
+         !> [optional] Can A data be overwritten and destroyed?
+         logical(lk), optional, intent(in) :: overwrite_a
+         !> [optional] state return flag. On error if not requested, the code will stop
+         type(linalg_state_type), optional, intent(out) :: err
+
+         !> Local variables
+         type(linalg_state_type) :: err0
+         integer(ilp) :: m,n,lda,ldu,ldv,info,k,lwork,lrwork,neig
+         logical(lk) :: copy_a
+         character :: task_u,task_v
+         ${rt}$, target :: work_dummy(1),u_dummy(1,1),v_dummy(1,1)
+         ${rt}$, allocatable :: work(:)
+         real(${rk}$), allocatable :: rwork(:)
+         ${rt}$, pointer :: amat(:,:),lreal(:),limag(:),umat(:,:),vmat(:,:)
+
+         !> Matrix size
+         m    = size(a,1,kind=ilp)
+         n    = size(a,2,kind=ilp)
+         k    = min(m,n)
+         neig = size(lambda,kind=ilp) 
+         lda  = m
+
+         if (.not.(k>0 .and. m==n)) then
+            err0 = linalg_state_type(this,LINALG_VALUE_ERROR,&
+                                          'invalid or matrix size a=',[m,n],', must be square.')
+            call linalg_error_handling(err0,err)
+            return
+         elseif (.not.neig>=k) then
```

Comment:
My coding practice is that checking for `.not.verified` is always better because it includes the edge cases (e.g. what if `neig` is `NaN` because we provided an unallocated array? 
But, we're talking about tiny details, so I'm also ok with checking `if (verified)`!

Reply:
If what is written here https://fortran-lang.discourse.group/t/how-to-fill-an-integer-array-with-some-nans/3696 is correct, an integer cannot be `NaN`. What might cause issues is that this query `neig = size(lambda,kind=ilp)` can return `1` with some compilers if the allocatable has not been allocated. To be in the safe side I would actually change in line 180 for
```fortran
neig = 0
if(allocated(lambda)) neig = size(lambda,kind=ilp) 
```